src/defs.h

```diff
@@ -682,11 +682,14 @@ struct finfo {
 	bool deleted;
 	struct {
 		unsigned int major, minor;
+		union {
+			int tty_index;
+		};
```

Comment:
A union with a single member?  Why?

Reply:
I assumed we would add more device-specific members in the future.
`union` may be the hint for potential contributors; it tells the person this is where to add a device-specific member.

Should we use `union` when we need it?

Reply:
Of course we can use union, I just didn't see the point in using it here, I had no idea you're expecting more device-specific members in the future.